addons/im_livechat/static/tests/thread_model_patch.test.js

```diff
@@ -36,3 +40,33 @@ test("Thread name unchanged when inviting new users", async () => {
     await contains(".o-discuss-ChannelMember", { text: "James" });
     await contains(".o-mail-Discuss-threadName[title='Visitor #20']");
 });
+
+test("Display livechat custom username if defined", async () => {
+    const pyEnv = await startServer();
+    pyEnv["res.partner"].write(serverState.partnerId, {
+        user_livechat_username: "livechat custom username",
+    });
+    const guestId = pyEnv["mail.guest"].create({ name: "Visitor #20" });
+    const channelId = pyEnv["discuss.channel"].create({
+        anonymous_name: "Visitor #20",
+        channel_member_ids: [
+            Command.create({ partner_id: serverState.partnerId }),
+            Command.create({ guest_id: guestId }),
+        ],
+        channel_type: "livechat",
+        livechat_operator_id: serverState.partnerId,
+    });
+    const env = await start();
+    rpc = rpcWithEnv(env);
+    await openDiscuss(channelId);
+    rpc("/mail/message/post", {
```

Comment:
We should stay as close as possible as the real world scenario. Sending RPCs like this one can easily lead to race conditions thus non deterministic tests. I'd suggest we type some text in the composer and type enter instead :)

Reply:
Done